keyboards/kiwikey/kawii9v2/lib/oled_custom_api.c

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 KiwiKey
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "oled_custom_api.h"
+
+extern uint8_t oled_buffer[OLED_MATRIX_SIZE];
+extern OLED_BLOCK_TYPE oled_dirty;
+#define OLED_ALL_BLOCKS_MASK (((((OLED_BLOCK_TYPE)1 << (OLED_BLOCK_COUNT - 1)) - 1) << 1) | 1)
+
+// x range: 0..127
+// y range: 0..7
+void oled_pan_area(char start_x, char end_x, char start_y, char end_y, bool left) {
+    uint16_t i = 0;
+    for (uint16_t y = start_y; y <= end_y; y++) {
+        if (left) {
+            for (uint16_t x = start_x; x < end_x; x++) {
+                i              = y * OLED_DISPLAY_WIDTH + x;
+                oled_buffer[i] = oled_buffer[i + 1];
+            }
+        } else {
+            for (uint16_t x = end_x; x > start_x; x--) {
+                i              = y * OLED_DISPLAY_WIDTH + x;
+                oled_buffer[i] = oled_buffer[i - 1];
+            }
+        }
+    }
+    oled_dirty = OLED_ALL_BLOCKS_MASK;
+}
```

Comment:
This may be a better way to handle it, as it accounts for rotation, and only dirties the rendered blocks.  This should be more compatible, and may be a bit faster.
```
extern oled_rotation_t oled_rotation;
extern uint8_t         oled_rotation_width;
extern uint8_t         oled_buffer[OLED_MATRIX_SIZE];
extern OLED_BLOCK_TYPE oled_dirty;

void oled_pan_section(bool left, uint16_t y_start, uint16_t y_end, uint16_t x_start, uint16_t x_end) {
    uint16_t i = 0;
    for (uint16_t y = y_start; y < y_end; y++) {
        if (left) {
            for (uint16_t x = x_start; x < x_end - 1; x++) {
                i              = y * oled_rotation_width + x;
                oled_buffer[i] = oled_buffer[i + 1];
                oled_dirty |= ((OLED_BLOCK_TYPE)1 << (i / OLED_BLOCK_SIZE));
            }
        } else {
            for (uint16_t x = x_end - 1; x >= x_start; x--) {
                i              = y * oled_rotation_width + x;
                oled_buffer[i] = oled_buffer[i - 1];
                oled_dirty |= ((OLED_BLOCK_TYPE)1 << (i / OLED_BLOCK_SIZE));
            }
        }
    }
}
```